core/trino-main/src/main/java/io/trino/cost/AggregationStatsRule.java

```diff
@@ -66,19 +67,22 @@ protected Optional<PlanNodeStatsEstimate> doCalculate(AggregationNode node, Stat
     public static PlanNodeStatsEstimate groupBy(PlanNodeStatsEstimate sourceStats, Collection<Symbol> groupBySymbols, Map<Symbol, Aggregation> aggregations)
     {
         PlanNodeStatsEstimate.Builder result = PlanNodeStatsEstimate.builder();
-        for (Symbol groupBySymbol : groupBySymbols) {
-            SymbolStatsEstimate symbolStatistics = sourceStats.getSymbolStatistics(groupBySymbol);
-            result.addSymbolStatistics(groupBySymbol, symbolStatistics.mapNullsFraction(nullsFraction -> {
-                if (nullsFraction == 0.0) {
-                    return 0.0;
-                }
-                return 1.0 / (symbolStatistics.getDistinctValuesCount() + 1);
-            }));
+        if (groupBySymbols.isEmpty()) {
+            result.setOutputRowCount(1);
```

Comment:
theoretically, you can have many empty grouping sets e.g.
```
trino:tpch> select count(*) from orders group by grouping sets ((), (), ());
  _col0   
----------
 15000000 
 15000000 
 15000000 
```
so maybe set the `outputRowCount` to the number of empty grouping sets.

Reply:
Final aggregation receives grouping key, there is possible work around, but this could complicate code for such artificial example (maybe you could provide some more realistic one?)

Reply:
you can use `AggregationNode#getGlobalGroupingSets`. should not be that complicated